src/dvc_objects/db.py

```diff
@@ -28,9 +29,9 @@ def wrap_iter(iterable, callback):
 
 
 class ObjectDB:
-    def __init__(self, fs: "FileSystem", path: str, **config):
+    def __init__(self, fs: "FileSystem", path: Optional[str] = None, **config):
         self.fs = fs
-        self.path = path
+        self.path = path if path is not None else fs.root_marker
```

Comment:
Is this really needed? It is very odd to make `path` optional.

Reply:
Usually, we may just want the path to be a root marker, no? Even if not, I find it good to default to, makes testing easier.

Reply:
Not really. And we would expect any user of this to set a path explicitly. Using root might be dangerous. Let's not change this, especially since this is not realated to add_fobj/bytes at all